<?xml version="1.0"?>
<robot name="single_leg">
  <gazebo>
    <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
      <robotNamespace>/single_leg</robotNamespace>
    </plugin>
  </gazebo>

  <link name="base">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.001 0.001 0.001"/>
      </geometry>
    </visual>
  </link>

  <joint name="world_to_trunk" type="floating">
    <origin xyz="0.0 0.0 0.0"/>
    <parent link="base"/>
    <child link="trunk"/>
  </joint>

  <link name="trunk">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://ilqr_single_leg/meshes/trunk.dae" scale="0.5 0.5 0.5"/>
      </geometry>
      <material name="orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.647 0.15 0.112"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.008465 0.0 -0.000763"/>
      <!-- <mass value="11.041"/> -->
      <mass value="3.041"/>
      <!-- <inertia ixx="0.050874" ixy="-0.000451628" ixz="0.000487603" iyy="0.64036" iyz="4.8356e-05" izz="0.65655"/> -->
      <inertia ixx="0.050874" ixy="0.0" ixz="0.0" iyy="0.64036" iyz="0.0" izz="0.65655"/>
    </inertial>
  </link>

  <joint name="thigh_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0.0 0"/>
    <parent link="trunk"/>
    <child link="thigh"/>
    <axis xyz="0 1 0"/>
    <dynamics damping="0" friction="0"/>
    <limit effort="35.278" lower="-1000" upper="1000" velocity="10000"/>
  </joint>

  <link name="thigh">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://ilqr_single_leg/meshes/thigh_mirror.dae" scale="1.0 1.0 1.0"/>
      </geometry>
      <material name="orange"/>
    </visual>
    <collision>
      <origin rpy="0 1.5707963267948966 0" xyz="0 0 -0.125"/>
      <geometry>
        <box size="0.25 0.0374 0.043"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.005607 0.0 -0.048199"/>
      <mass value="0.639"/>
      <!-- <inertia ixx="0.005666803" ixy="-3.597e-06" ixz="0.000491446" iyy="0.005847229" iyz="-1.0086e-05" izz="0.000369811"/> -->
      <inertia ixx="0.005666803" ixy="0.0" ixz="0.0" iyy="0.005847229" iyz="0.0" izz="0.000369811"/>
    </inertial>
  </link>

  <joint name="calf_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.25"/>
    <parent link="thigh"/>
    <child link="calf"/>
    <axis xyz="0 1 0"/>
    <dynamics damping="0" friction="0"/>
    <limit effort="44.4" lower="-1000" upper="1000" velocity="10000"/>
  </joint>

  <link name="calf">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://ilqr_single_leg/meshes/calf.dae" scale="1 1 1"/>
      </geometry>
      <material name="orange"/>
    </visual>
    <collision>
      <origin rpy="0 1.5707963267948966 0" xyz="0 0 -0.125"/>
      <geometry>
        <box size="0.25 0.0208 0.016"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.002781 0 -0.142518"/>
      <mass value="0.207"/>
      <!-- <inertia ixx="0.006341369" ixy="-3e-09" ixz="-8.7951e-05" iyy="0.006355157" iyz="-1.336e-06" izz="3.9188e-05"/> -->
      <inertia ixx="0.006341369" ixy="0.0" ixz="0.0" iyy="0.006355157" iyz="0.0" izz="3.9188e-05"/>
    </inertial>
  </link>

  <joint name="foot_fixed" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 -0.25"/>
    <parent link="calf"/>
    <child link="foot"/>
  </joint>

  <link name="foot">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.0165"/>
      </geometry>
      <material name="green"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.0265"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.06"/>
      <inertia ixx="1.6853999999999998e-05" ixy="0.0" ixz="0.0" iyy="1.6853999999999998e-05" iyz="0.0" izz="1.6853999999999998e-05"/>
    </inertial>
  </link>
  <!-- <transmission name="hip_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="hip_joint">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="hip_motor">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission> -->

  <transmission name="thigh_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="thigh_joint">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="thigh_motor">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="calf_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="calf_joint">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="calf_motor">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
</robot>


